### PR TITLE
Make it work on both of Linux and Mac OS

### PR DESCRIPTION
**Pull request by fukamachi**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,14 @@ CC = gcc
 AS = $(CC)
 LD = $(CC)
 RM = rm -f
+SYSTEM=$(shell uname 2>/dev/null)
+ifeq ($(SYSTEM), Darwin)
+SHARED_OBJ = libbcrypt.$(VERSION).dylib
+SHAREDFLAGS = -shared -W1,-soname,$(SHARED_OBJ)
+else
+SHARED_OBJ = libbcrypt.$(VERSION).so
+SHAREDFLAGS = -shared -Wl,-soname,$(SHARED_OBJ)
```
Comment (rudolph-miller) [inline Makefile:18]: There isn't -soname option with darwin, maybe.
You can use -install_name instead of -soname.

Comment (rudolph-miller) [inline Makefile:18]: Sorry, it's on the 15th line.

Comment (fukamachi) [inline Makefile:18]: I don't think so, but I replaced it anyway.

https://gist.githubusercontent.com/fukamachi/0a720525eda20a6ba53a/raw/2d9ad235262bd38fdc963bb975edb1c9bb8075d3/libbcrypt.rb

**Comment by rudolph-miller** [comment]
thx for adding formula.

**Pull request merged**